x/config/config.go

```diff
@@ -68,39 +69,88 @@ func WrapStreamDialer(dialer transport.StreamDialer, transportConfig string) (tr
 	return dialer, nil
 }
 
-func newStreamDialerFromPart(innerDialer transport.StreamDialer, oneDialerConfig string) (transport.StreamDialer, error) {
+type StreamDialer struct {
+	transport.StreamDialer
+	Config string
+}
+
```

Comment:
Let's define a consistent string. Also, make it stand out more as a placeholder.
```suggestion
const redactedPlaceholder = "[REDACTED]"

```